### PR TITLE
log: Send extra key values to sentry in ErrorWithSentry

### PR DESCRIPTION
**Pull request by fishy**
Also move ErrorWithSentry from log.go to sentry.go.

### CODE REVIEW

File: log/sentry.go [hunk 1]
```diff
@@ -88,3 +89,45 @@ func (c closer) Close() error {
...
+			// There's a dangling key, log and append an empty value to avoid panic.
+			Errorw(
+				"Dangling key in ErrorWithSentry",
+				"keysAndValues", keysAndValues,
+			)
+			keysAndValues = append(keysAndValues, "")
+		}
+		hub = hub.Clone()
+		hub.ConfigureScope(func(scope *sentry.Scope) {
+			for i := 0; i < len(keysAndValues); i++ {
```
Comment (pacejackson) [inline log/sentry.go:120]: Shouldn't this be `i += 2`?

```suggestion
			for i := 0; i < len(keysAndValues); i+=2 {
```

Comment (fishy) [reply log/sentry.go:120]: Yes 🤦

Comment (pacejackson) [reply log/sentry.go]: Easy mistake to make 😅

**Comment by fishy** [comment]
:haircut: Changed to ignore zap.Field in ErrorWithSentry instead of break it

**Comment by fishy** [comment]
💇 Added tests because this is a place that's really easy to make mistakes 😬

**Pull request merged**